### ISSUE TITLE
Printing fails

### ISSUE THREAD

**Issue by apocalypxx** [brave/browser-laptop#1616]
![image](https://cloud.githubusercontent.com/assets/18390479/15091525/edcb92d0-147d-11e6-8522-753359fa9b92.png)

Pdf printing (with pdf995) fails. Process is stuck at spooling

**Comment by bbondy** [comment]
Does the same process work in Chrome for you?

**Comment by apocalypxx** [comment]
Yes it does
​

**Comment by apocalypxx** [comment]
Yes it works well (please see previous mail with attachments).
With Brave, the error seems to be the printer seems to get stuck in the
spooling state.

**Comment by luixxiul** [comment]
Confirmed on Windows 10. While "Foxit Reader PDF Printer" create a file, others like "Microsoft Print to PDF" neither create nor spool it and nothing happens.

**Comment by luixxiul** [comment]
reported on support
> Seems like the process gets stuck in "spooling". Stuck as in really stuck and locked down. 
System Windows 10, 64x.

**Comment by luixxiul** [comment]
Do we have this? https://github.com/electron/electron/blob/master/docs/api/web-contents.md#webcontentsprinttopdfoptions-callback

**Comment by luixxiul** [comment]
from support

> When I send a print job while working in Brave, the printer just says "spooling" and nothing happens.
> 
> I thought there was something wrong w/my printer but I got the bright idea to try the same job with Firefox & Edge, and I can print.

SquirrelSetup.log: 

````
2016-05-11 22:15:28> Program: Starting Squirrel Updater: --install .
2016-05-11 22:15:29> Program: Starting install, writing to C:\Users\Diane\AppData\Local\SquirrelTemp
2016-05-11 22:15:33> Program: About to install to: C:\Users\Diane\AppData\Local\Brave
2016-05-11 22:15:36> CheckForUpdateImpl: Failed to load local releases, starting from scratch: System.IO.DirectoryNotFoundException: Could not find a part of the path 'C:\Users\Diane\AppData\Local\Brave\packages\RELEASES'.
   at System.IO.__Error.WinIOError(Int32 errorCode, String maybeFullPath)
   at System.IO.FileStream.Init(String path, FileMode mode, FileAccess access, Int32 rights, Boolean useRights, FileShare share, Int32 bufferSize, FileOptions options, SECURITY_ATTRIBUTES secAttrs, String msgPath, Boolean bFromProxy, Boolean useLongPath, Boolean checkHost)
   at System.IO.FileStream..ctor(String path, FileMode mode, FileAccess access, FileShare share)
   at Squirrel.Utility.LoadLocalReleases(String localReleaseFile)
   at Squirrel.UpdateManager.CheckForUpdateImpl.<CheckForUpdate>d__2.MoveNext()
2016-05-11 22:15:37> CheckForUpdateImpl: Reading RELEASES file from C:\Users\Diane\AppData\Local\SquirrelTemp
2016-05-11 22:15:40> CheckForUpdateImpl: First run or local directory is corrupt, starting from scratch
2016-05-11 22:15:40> ApplyReleasesImpl: Writing files to app directory: C:\Users\Diane\AppData\Local\Brave\app-0.9.4
2016-05-11 22:16:01> ApplyReleasesImpl: Moving file C:\Users\Diane\AppData\Local\Brave\app-0.9.4\lib\net45\atom_resources.pak to C:\Users\Diane\AppData\Local\Brave\app-0.9.4\atom_resources.pak
2016-05-11 22:16:01> ApplyReleasesImpl: Moving file C:\Users\Diane\AppData\Local\Brave\app-0.9.4\lib\net45\Brave.exe to C:\Users\Diane\AppData\Local\Brave\app-0.9.4\Brave.exe
2016-05-11 22:16:01> ApplyReleasesImpl: Moving file C:\Users\Diane\AppData\Local\Brave\app-0.9.4\lib\net45\content_resources_200_percent.pak to C:\Users\Diane\AppData\Local\Brave\app-0.9.4\content_resources_200_percent.pak
2016-05-11 22:16:01> ApplyReleasesImpl: Moving file C:\Users\Diane\AppData\Local\Brave\app-0.9.4\lib\net45\content_shell.pak to C:\Users\Diane\AppData\Local\Brave\app-0.9.4\content_shell.pak
2016-05-11 22:16:01> ApplyReleasesImpl: Moving file C:\Users\Diane\AppData\Local\Brave\app-0.9.4\lib\net45\d3dcompiler_47.dll to C:\Users\Diane\AppData\Local\Brave\app-0.9.4\d3dcompiler_47.dll
2016-05-11 22:16:01> ApplyReleasesImpl: Moving file C:\Users\Diane\AppData\Local\Brave\app-0.9.4\lib\net45\extensions_api_resources.pak to C:\Users\Diane\AppData\Local\Brave\app-0.9.4\extensions_api_resources.pak
2016-05-11 22:16:01> ApplyReleasesImpl: Moving file C:\Users\Diane\AppData\Local\Brave\app-0.9.4\lib\net45\extensions_renderer_resources.pak to C:\Users\Diane\AppData\Local\Brave\app-0.9.4\extensions_renderer_resources.pak
2016-05-11 22:16:01> ApplyReleasesImpl: Moving file C:\Users\Diane\AppData\Local\Brave\app-0.9.4\lib\net45\extensions_resources.pak to C:\Users\Diane\AppData\Local\Brave\app-0.9.4\extensions_resources.pak
2016-05-11 22:16:01> ApplyReleasesImpl: Moving file C:\Users\Diane\AppData\Local\Brave\app-0.9.4\lib\net45\ffmpeg.dll to C:\Users\Diane\AppData\Local\Brave\app-0.9.4\ffmpeg.dll
2016-05-11 22:16:01> ApplyReleasesImpl: Moving file C:\Users\Diane\AppData\Local\Brave\app-0.9.4\lib\net45\icudtl.dat to C:\Users\Diane\AppData\Local\Brave\app-0.9.4\icudtl.dat
2016-05-11 22:16:01> ApplyReleasesImpl: Moving file C:\Users\Diane\AppData\Local\Brave\app-0.9.4\lib\net45\libGLESv2.dll to C:\Users\Diane\AppData\Local\Brave\app-0.9.4\libGLESv2.dll
2016-05-11 22:16:01> ApplyReleasesImpl: Moving file C:\Users\Diane\AppData\Local\Brave\app-0.9.4\lib\net45\LICENSE to C:\Users\Diane\AppData\Local\Brave\app-0.9.4\LICENSE
2016-05-11 22:16:01> ApplyReleasesImpl: Moving file C:\Users\Diane\AppData\Local\Brave\app-0.9.4\lib\net45\libEGL.dll to C:\Users\Diane\AppData\Local\Brave\app-0.9.4\libEGL.dll
2016-05-11 22:16:01> ApplyReleasesImpl: Moving file C:\Users\Diane\AppData\Local\Brave\app-0.9.4\lib\net45\msvcr120.dll to C:\Users\Diane\AppData\Local\Brave\app-0.9.4\msvcr120.dll
2016-05-11 22:16:01> ApplyReleasesImpl: Moving file C:\Users\Diane\AppData\Local\Brave\app-0.9.4\lib\net45\msvcp120.dll to C:\Users\Diane\AppData\Local\Brave\app-0.9.4\msvcp120.dll
2016-05-11 22:16:01> ApplyReleasesImpl: Moving file C:\Users\Diane\AppData\Local\Brave\app-0.9.4\lib\net45\natives_blob.bin to C:\Users\Diane\AppData\Local\Brave\app-0.9.4\natives_blob.bin
2016-05-11 22:16:01> ApplyReleasesImpl: Moving file C:\Users\Diane\AppData\Local\Brave\app-0.9.4\lib\net45\squirrel.exe to C:\Users\Diane\AppData\Local\Brave\app-0.9.4\squirrel.exe
2016-05-11 22:16:01> ApplyReleasesImpl: Moving file C:\Users\Diane\AppData\Local\Brave\app-0.9.4\lib\net45\node.dll to C:\Users\Diane\AppData\Local\Brave\app-0.9.4\node.dll
2016-05-11 22:16:01> ApplyReleasesImpl: Moving file C:\Users\Diane\AppData\Local\Brave\app-0.9.4\lib\net45\ui_resources_200_percent.pak to C:\Users\Diane\AppData\Local\Brave\app-0.9.4\ui_resources_200_percent.pak
2016-05-11 22:16:01> ApplyReleasesImpl: Moving file C:\Users\Diane\AppData\Local\Brave\app-0.9.4\lib\net45\vccorlib120.dll to C:\Users\Diane\AppData\Local\Brave\app-0.9.4\vccorlib120.dll
2016-05-11 22:16:01> ApplyReleasesImpl: Moving file C:\Users\Diane\AppData\Local\Brave\app-0.9.4\lib\net45\snapshot_blob.bin to C:\Users\Diane\AppData\Local\Brave\app-0.9.4\snapshot_blob.bin
2016-05-11 22:16:01> ApplyReleasesImpl: Moving file C:\Users\Diane\AppData\Local\Brave\app-0.9.4\lib\net45\xinput1_3.dll to C:\Users\Diane\AppData\Local\Brave\app-0.9.4\xinput1_3.dll
2016-05-11 22:16:06> ApplyReleasesImpl: Squirrel Enabled Apps: [C:\Users\Diane\AppData\Local\Brave\app-0.9.4\Brave.exe]
2016-05-11 22:16:18> ApplyReleasesImpl: Starting fixPinnedExecutables
2016-05-11 22:16:18> ApplyReleasesImpl: Examining Pin: EPSON Scan.lnk
2016-05-11 22:16:18> ApplyReleasesImpl: Examining Pin: File Explorer.lnk
2016-05-11 22:16:18> ApplyReleasesImpl: Examining Pin: Google Chrome Canary.lnk
2016-05-11 22:16:18> ApplyReleasesImpl: Examining Pin: Google Drive.lnk
2016-05-11 22:16:18> ApplyReleasesImpl: Examining Pin: Mozilla Firefox.lnk
2016-05-11 22:16:18> ApplyReleasesImpl: Examining Pin: OpenOffice Writer.lnk
2016-05-11 22:16:18> ApplyReleasesImpl: Examining Pin: OpenOffice.lnk
2016-05-11 22:16:18> ApplyReleasesImpl: Examining Pin: Picasa 3.lnk
2016-05-11 22:16:18> ApplyReleasesImpl: Examining Pin: Skype for desktop.lnk
2016-05-11 22:16:18> ApplyReleasesImpl: Fixing up tray icons
2016-05-11 22:16:18> ApplyReleasesImpl: cleanDeadVersions: for version 0.9.4
2016-05-11 22:16:18> ApplyReleasesImpl: cleanDeadVersions: exclude folder app-0.9.4
````

**Comment by bsclifton** [comment]
I'd like to +1 this

Before leaving for a trip, I tried printing our tickets. I troubleshooter for a good 10 mins before trying Chrome and having it work there (same printer) on the first try

**Comment by luixxiul** [comment]
On master branch, Ubuntu 14.10, the printing in PDF succeeds (eg. [output.pdf](https://github.com/brave/browser-laptop/files/287677/output.pdf)). Maybe specific to Windows?

**Comment by freshcut1** [comment]
I have tried to trouble-shoot this problem every way I can think of. Print functions on webpages and browser buttons work fine in Firefox and MS Edge. In Brave, I get nothing, no error messages or any print output. Otherwise, I could send a screenshot. It is a particular problem for me, as I travel a lot and constantly make and print reservation info and boarding passes. I can't fully commit to Brave until we figure this out

**Comment by bradleyrichter** [comment]
mac OS prints to PDF but page breaks are not what you would expect:

![image](https://cloud.githubusercontent.com/assets/13509546/15901967/795a7fc4-2d5a-11e6-8810-1891c373e926.png)

**Comment by luixxiul** [comment]
+1 from support

**Comment by kazahjelly** [comment]
This issue occurs, on Windows 10 printing does not occur for any web page. My solution at the moment is to use an alternative web browser for printing.

**Comment by srirambv** [comment]
Print option is not enabled for Nitro PDF printer. 
![print](https://cloud.githubusercontent.com/assets/17010094/16837960/87479772-49be-11e6-9f59-3a5950de9daf.png)

**Comment by Mordineus** [comment]
Exact same problem Windows 7

**Comment by bbondy** [comment]
Fixed here for 0.11.4dev
https://github.com/brave/electron/commit/a07c7cf08062834d4134ba6ef0c2f198978e782c
